tests/suggestion-box.test.ts

```diff
@@ -0,0 +1,309 @@
+                    .listkey=${"name"}
+                >
+                    <div slot="sgn-item-template">{name}</div>
+                </aalam-sgn-box>
+            `
+        );
+
+        await el.updateComplete;
+
+        const input = el.shadowRoot?.querySelector("input");
+        if (!input) throw new Error("Input element not found");
+
+        input.focus();
+        await triggerFocusFor(input);
+
+        // Simulate typing in the input
+        input.value = "test";
+        input.dispatchEvent(
+            new Event("input", { bubbles: true, composed: true })
+        );
+        await el.updateComplete;
+
+        // Simulate pressing ArrowDown key
+        const arrowDownEvent = new KeyboardEvent("keydown", {
+            key: "ArrowDown",
+            bubbles: true,
+            composed: true,
+        });
+        input.dispatchEvent(arrowDownEvent);
+        await el.updateComplete;
+
+        const enterEvent = new KeyboardEvent("keydown", {
+            key: "Enter",
+            bubbles: true,
+            composed: true,
+        });
+        input.dispatchEvent(enterEvent);
+        await el.updateComplete;
+
+        expect(input.value).to.equal("test1");
+        expect(el.show_container).to.be.false;
+    });
+
+    it("should close container on outside click", async () => {
+        const el = await fixture(html`<aalam-sgn-box></aalam-sgn-box>`);
+        const input = el.shadowRoot?.querySelector("input");
+        const container = el.shadowRoot?.querySelector("#sgn-container");
+
+        expect(container?.style.display).to.equal("none");
+
+        input?.focus();
+        await el.updateComplete;
+
+        expect(container?.style.display).to.equal("block");
+
+        document.body.click();
+        await el.updateComplete;
+
+        expect(container?.style.display).to.equal("none");
+    });
+
+    it("should load more suggestions on 'Loadmore' click", async () => {
+        const el = await fixture(
+            html`
+                <aalam-sgn-box>
+                    <div slot="sgn-item-template">{name}</div>
+                </aalam-sgn-box>
+            `
+        );
+
+        const input = el.shadowRoot?.querySelector("input");
+        if (!input) throw new Error("Input element not found");
+
+        input.focus();
+        input.value = "test";
+        input.dispatchEvent(
+            new Event("input", { bubbles: true, composed: true })
+        );
+        await el.updateComplete;
+
+        el.setSuggestion(
+            [
+                { name: "test1" },
+                { name: "test2" },
+                { name: "test3" },
+                { name: "test4" },
+                { name: "test5" },
+            ],
+            true
+        );
+        await el.updateComplete;
+        const loadmoreItem = el?.querySelector(".sgn-loadmore");
+
+        expect(el.private_items.length).to.equal(6); // Initially loaded 5, then added 1 more
+        loadmoreItem?.click();
```

Comment:
Check if the load more event is received, and do thte appendSuggestion in the load more event. Also check if the active item is the first element of the new list.